tests/datastreams/test_readers.py

```diff
@@ -133,4 +158,62 @@ def test_json_element_reader(json_element_file, json_element):
     assert count == 1
 
 
+API_JSON_RESPONSE_CONTENT = {
+    "hits": {
+        "hits": [
+            {
+                "files": [
+                    {
+                        "links": {
+                            "self": "https://example.com/api/records/12345678/files/filename.zip/content"
+                        }
+                    }
+                ]
+            }
+        ]
+    }
+}
+
```

Comment:
They all look like nice options. :)

One downside of snapshot-style is that the JSON will include a lot of irrelevant data.
For instance, here I kept only what was necessary for the test to pass (i.e. `hits.hits.files.links.self`) instead of the full JSON, and I used a fake link.